packages/std/src/msgpack.rs

```diff
@@ -0,0 +1,126 @@
+    T: Serialize + ?Sized,
+{
+    to_msgpack_vec(data).map(Binary::new)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde::Deserialize;
+
+    #[derive(Serialize, Deserialize, Debug, PartialEq)]
+    #[serde(rename_all = "snake_case")]
+    enum SomeMsg {
+        Refund {},
+        ReleaseAll {
+            image: String,
+            amount: u32,
+            time: u64,
+            karma: i32,
+        },
+        Cowsay {
+            text: String,
+        },
+    }
+
+    fn refund_test_vector() -> (SomeMsg, &'static [u8]) {
+        let msg = SomeMsg::Refund {};
+        let serialized = &[129, 166, 114, 101, 102, 117, 110, 100, 144];
+        (msg, serialized)
+    }
+
+    fn release_all_test_vector() -> (SomeMsg, &'static [u8]) {
+        let msg = SomeMsg::ReleaseAll {
+            image: "foo".to_string(),
+            amount: 42,
+            time: 18446744073709551615,
+            karma: -17,
+        };
+        let serialized = &[
+            129, 171, 114, 101, 108, 101, 97, 115, 101, 95, 97, 108, 108, 148, 163, 102, 111, 111,
+            42, 207, 255, 255, 255, 255, 255, 255, 255, 255, 239,
+        ];
+        (msg, serialized)
+    }
+
+    fn special_chars_test_vector() -> (SomeMsg, &'static [u8]) {
+        let msg = SomeMsg::Cowsay {
+            text: "foo\"bar\\\"bla".to_string(),
+        };
+        let serialized = &[
+            129, 166, 99, 111, 119, 115, 97, 121, 129, 164, 116, 101, 120, 116, 172, 102, 111, 111,
+            34, 98, 97, 114, 92, 34, 98, 108, 97,
+        ];
+        (msg, serialized)
+    }
+
+    #[test]
+    fn to_msgpack_vec_works() {
+        let (msg, expected) = refund_test_vector();
+        let serialized = to_msgpack_vec(&msg).unwrap();
+        assert_eq!(serialized, expected);
+
+        let (msg, expected) = release_all_test_vector();
+        let serialized = to_msgpack_vec(&msg).unwrap();
+        assert_eq!(serialized, expected);
+    }
+
+    #[test]
+    fn from_msgpack_works() {
+        let (msg, serialized) = refund_test_vector();
+        let deserialized: SomeMsg = from_msgpack(serialized).unwrap();
+        assert_eq!(deserialized, msg);
+
+        let (msg, serialized) = release_all_test_vector();
+        let deserialized: SomeMsg = from_msgpack(serialized).unwrap();
+        assert_eq!(deserialized, msg);
+    }
+
+    #[test]
+    fn from_msgpack_or_binary() {
+        let msg = SomeMsg::Refund {};
+        let serialized: Binary = to_msgpack_binary(&msg).unwrap();
+
+        let parse_binary: SomeMsg = from_msgpack(&serialized).unwrap();
+        assert_eq!(parse_binary, msg);
+
+        let parse_slice: SomeMsg = from_msgpack(serialized.as_slice()).unwrap();
+        assert_eq!(parse_slice, msg);
+    }
+
+    #[test]
+    fn from_msgpack_works_for_special_chars() {
+        let (msg, serialized) = special_chars_test_vector();
+        let deserialized: SomeMsg = from_msgpack(serialized).unwrap();
+        assert_eq!(deserialized, msg);
+    }
+}
```

Comment:
Could you test the properties we discussed recently:
1. Deserializing into a type with different field order is possible
2. Deserializing into a type with additional fields is possible

Reply:
Added this to my TODO list. I'll get on it Wed/Thur

Reply:
Alright, so:
1. not supported
2. works as expected, but you can only add to the end of the struct

Reply:
Even with https://docs.rs/rmp-serde/latest/rmp_serde/config/struct.StructMapConfig.html this is not possible?

Reply:
Good catch, I forgot to change it. Done now, both points work.